src/PgsqlServer.cpp

```diff
@@ -0,0 +1,222 @@
+    std::atomic_int& workersCounter = *arg->workersCounter;
+
+    defer(workersCounter--); // already incremented in the parent process
+
+    Socket socket(arg->socket);
+    PgsqlWorker worker(socket);
+    delete arg;
+
+    /* No other thread is going to join() this one */
+    pthread_detach(pthread_self());
+
+    worker.run();
+
+    pthread_exit(nullptr);
+}
+
+PgsqlWorker::PgsqlWorker(Socket& socket)
+  : _socket(socket){
+}
+
+size_t PgsqlWorker::getMessageSize(){
+    char char_size[4];
+    _socket.read(char_size, sizeof(char_size));
+
+    uint32_t size = ntohl(*(uint32_t*)&char_size);
+    size -= sizeof(char_size);
+
+    return (size_t) size;
+}
+
+void PgsqlWorker::sendParameter(const char* field, const char * parameter, const char* type) {
+    uint32_t message_size = htonl(
+                                    strlen(type) + 
+                                    sizeof(uint32_t) + 
+                                    strlen(field) + 
+                                    delim_len + 
+                                    strlen(parameter)
+                                );
+
+    _socket.write(type, strlen(type));
+    _socket.write((char*) &message_size, sizeof(message_size));
+    _socket.write(field, strlen(field));
+    _socket.write((char*) delimeter, delim_len);
+    _socket.write(parameter, strlen(parameter));
+}
+
+void PgsqlWorker::sendServerConfiguration(){
+        _socket.write((char*) "R", 1);
+
+        //Not meaning space
+        unsigned char space[] = {0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00};
+
+        _socket.write((char*) space,sizeof(space));
+
+        sendParameter("application_name", "", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("client_encoding", "UTF8", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("DateStyle", "ISO, MDY", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("integer_datetimes", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("IntervalStyle", "postgres", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("is_superuser", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("server_encoding", "UTF8", "S");
+        
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("server_version", "9.6.5", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("session_authorization", "postgres", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("standard_conforming_strings", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("TimeZone", "W-SU", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        _socket.write((char*) "K", 1);
+        
+        //Not meaning end of configuration message
```

Comment:
This is a weird comment. It doesn't tell me a lot - so end[] is not meaning something, but what it means? I suggest to rewrite it.